conda/install.py

```diff
@@ -922,6 +927,8 @@ def move_path_to_trash(path):
         except OSError as e:
             log.debug("Could not move %s to %s (%s)" % (path, trash_dir, e))
         else:
+            if path in linked_data_:
+                del linked_data_[path]
```

Comment:
Note the function `delete_linked_data` may be useful here. It accepts a `prefix` and a `dist` and removes a record from `linked_data_`, *and the corresponding data on disk in the `conda-meta` directory as well.*